File: Recursion_Array/max.java
import java.util.Scanner;
public class max
{
    public static Scanner scn = new Scanner(System.in);
    public static int maxElement(int[] arr, int idx)
    {
        if(idx == arr.length)
        {
            return -(int)1e8;
        }
        int ans = maxElement(arr, idx + 1);
        return Math.max(arr[idx], ans);
    }

    public static void input(int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            arr[i] = scn.nextInt();
        }
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        System.out.println(maxElement(arr, 0));
    }
}

File: Recursion_Array/max.cpp
#include<iostream>
#include<vector>
using namespace std;

int maxArray(vector<int>& arr, int idx)
{
    if(idx == arr.size())
    {
        return -1e8;
    }
    int maxOfSmallSeg = maxArray(arr, idx +1);
    return max(arr[idx], maxOfSmallSeg);
}

void input(vector<int>& arr)
{
    for(int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    input(arr);
    cout << maxArray(arr, 0);
}